tap_outreach/sync.py

```diff
@@ -250,11 +250,6 @@ def process_records(stream, mdata, max_modified, records, filter_field, fks):
                             raise Exception(
                                 'null or `id` field expected for `data` relationship')
 
-                        # potential fix for the issue - https://github.com/singer-io/tap-outreach/issues/20
-                        if stream.tap_stream_id not in ["prospects", "events"]:
-                            if fk_field_name in record_flat:
-                                raise Exception('`{}` exists as both an attribute and generated relationship name'.format(fk_field_name))
-
                         if data_value is None:
                             record_flat[fk_field_name] = None
```

Comment:
Should we still be overwriting the attribute with None if there is no relationship?

Reply:
Outreach support responded us with this : 
```
Thank you for reaching out to the Outreach Platform team!
I would recommend relying on the relationship since as per our [documentation](https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Furldefense.com%2Fv3%2F__https%3A%2F%2Fdevelopers.outreach.io%2Fapi%2Freference%2Ftag%2FProspect%2F__%3B!!CiXD_PY!UTT9dqhfRjVE5iyek_cXxH1yZFSq5zzjoQsXK1A8Pn6A-j-L_ukNDg4_eY68RpVafM_4DKExoUC0zzTJeA%24&data=05%7C01%7CSourabh.Gandhi%40qlik.com%7C28327603cd0242d70b3e08dbc6eae479%7Cc21eeb5ff5a644e8a997124f2f7a497c%7C0%7C0%7C638322484406305361%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=hFY0Q2NOiXjtRgUVGAnJU6jhqyaWIXd%2BNHdQOzXcBzY%3D&reserved=0) that is the one documented.
The attribute is not documented, which means, it can be removed at any time.
Please let me know if there is anything else I can help you with.
Thank you!
```

Reply:
After making API call for **/prospects** API, we got a value for **updaterId** attribute, but in the relationship we're getting `null`. And, as per Outreach support they give priority to relationship over the attribute. Hence, we may need to keep this code and explicitly set it to `None`.

Reply:
We should only prioritize the relationship value if we have a relationship value.

Reply:
In their [documentation](https://developers.outreach.io/api/reference/tag/Prospect/) the attribute **updaterId** is not even documented. If we don't set the field value as per relationship, we may get the wrong data which is not there. 
So, we should still proceed with removing this code which explicitly sets the relationship field value to `None` ?

Reply:
Outreach supports custom fields. If we have a value for a custom field and no value for the relationship then we should extract the custom field's value.